src/Style.tsx

```diff
@@ -78,17 +79,19 @@ export const Style = forwardRef<HTMLLinkElement | HTMLStyleElement, Props>(
     const style = /*#__NOINLINE__*/ useStyle(props.children, options);
 
     if (options.hasSourceMap) {
-      return createElement(Link, {
-        passedProps,
-        ref: ref as MutableRefObject<HTMLLinkElement>,
-        style,
-      });
+      return createPortal(
```

Comment:
By rendering these in a Portal, it allows React to handle tracing updates as well as passing through the `ref` value seamlessly.